src/FluentValidation/Internal/PropertyRule.cs

```diff
@@ -206,11 +200,9 @@ TProperty PropertyFunc(T instance)
 					InvokePropertyValidator(context, accessor, propertyName, validator);
 				}
 
-				// If there has been at least one failure, and our CascadeMode has been set to StopOnFirst
+				// If there has been at least one failure, and our CascadeMode has been set to Stop
 				// then don't continue to the next rule
-#pragma warning disable 618
-				if (context.Failures.Count > totalFailures && (cascade == CascadeMode.StopOnFirstFailure || cascade == CascadeMode.Stop)) {
-#pragma warning restore 618
+				if (context.Failures.Count > totalFailures && CascadeMode == CascadeMode.Stop) {

```

Comment:
Related to above, this should pull from the local `cascadeMode` variable, not from the property